src/OrchardCore/OrchardCore/Modules/ModularBackgroundService.cs

```diff
@@ -34,17 +36,25 @@ internal class ModularBackgroundService : BackgroundService
         private readonly IHttpContextAccessor _httpContextAccessor;
         private readonly ILogger _logger;
         private readonly IClock _clock;
+        private readonly bool _lazyBuildDisabled;
 
         public ModularBackgroundService(
             IShellHost shellHost,
             IHttpContextAccessor httpContextAccessor,
             ILogger<ModularBackgroundService> logger,
-            IClock clock)
+            IClock clock,
+            Microsoft.Extensions.Configuration.IConfiguration configuration)
```

Comment:
Shouldn't it be an options class?

Reply:
Yes, useful if multiple option fields and/or injected in multiple places. But not necessarily, we can get any config section "field" separately. But maybe still better to use an option class even if only one field and only used here. Note: Can be done later on without breaking change.

I'am also thinking about the section name `OC_Shells_LazyBuild` and its field `Disable`, maybe an `OC_Shells` section with a new `ShellsOptions` with a first `LazyBuild` field. Note: We already have a `ShellOptions` (for `App_Data` config) but only used at the tenant level, here it is at the host level.

Hmm, but maybe it is a background service option, e.g. if there is no bg task defined, even if you disable the lazy build option, a given shell will be still lazy built on a first request. So here it is to say to the bg service (if there are some bg tasks) that it can warmup a shell by executing a bg task.

I will try to think about it and suggest something this week end.